chain-evm/src/machine.rs

```diff
@@ -0,0 +1,211 @@
+//! # The Virtual Machine
+//!
+//! Abstractions for the EVM runtime.
+//!
+//! This module contains types that abstract away implementation details from the `evm` crate,
+//! specially those involving node and stack configurations, and runtime execution.
+//!
+//! ## Handler <- EVM Context Handler
+//! ## StackState<'config>
+//!
+
+use evm::{
+    backend::{Apply, ApplyBackend, Backend, Basic, Log, MemoryVicinity},
+    executor::StackState,
+    Context,
+};
+use primitive_types::{H160, H256, U256};
+
+use crate::state::AccountTrie;
+
+/// Environment values for the machine backend.
+pub type Environment = MemoryVicinity;
```

Comment:
Good naming, IDK why do the SputnikVM developers use "vicinity".

Reply:
Thanks, I got the name from reading the "yellow paper".